sdk/src/types/block/protocol.rs

```diff
@@ -107,32 +172,24 @@ impl ProtocolParameters {
         network_name_to_id(&self.network_name)
     }
 
-    /// Returns the bech32 HRP of the [`ProtocolParameters`].
-    pub fn bech32_hrp(&self) -> &Hrp {
-        &self.bech32_hrp
-    }
-
-    /// Returns the below max depth of the [`ProtocolParameters`].
-    pub fn below_max_depth(&self) -> u8 {
-        self.below_max_depth
+    /// Returns the mana decay factors slice of the [`ProtocolParameters`].
+    pub fn mana_decay_factors(&self) -> Option<&[u32]> {
+        self.mana_decay_factors.as_ref().map(|slice| slice.as_ref())
     }
 
-    /// Returns the rent structure of the [`ProtocolParameters`].
-    pub fn rent_structure(&self) -> &RentStructure {
-        &self.rent_structure
+    /// Returns the liveness threshold of the [`ProtocolParameters`].
+    pub fn liveness_threshold(&self) -> Option<SlotIndex> {
+        self.liveness_threshold
     }
 
-    /// Returns the token supply of the [`ProtocolParameters`].
-    pub fn token_supply(&self) -> u64 {
-        self.token_supply
+    /// Returns the epoch nearing threshold of the [`ProtocolParameters`].
+    pub fn epoch_nearing_threshold(&self) -> SlotIndex {
+        self.epoch_nearing_threshold
     }
 
-    pub fn genesis_unix_timestamp(&self) -> u32 {
-        self.genesis_unix_timestamp
-    }
-
-    pub fn slot_duration_in_seconds(&self) -> u8 {
-        self.slot_duration_in_seconds
+    pub fn slots_per_epoch_exponent(&self) -> u32 {
+        self.slots_per_epoch_exponent
+            .unwrap_or(DEFAULT_SLOTS_PER_EPOCH_EXPONENT)
     }
 
     pub fn slot_index(&self, timestamp: u64) -> SlotIndex {
```

Comment:
Can comment on the line but maybe `calc_slot_index` would be more interesting as a method on `SlotIndex` directly, `build` or whatever best suited